build_tools/python_deploy/write_requirements.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+# Copyright 2024 Advanced Micro Devices, Inc.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# This scripts writes the `packaging/shark-platform/requirements.txt` file
+# and pins the versions of the dependencies accordingly. For nighly releases,
+#  * sharktank
+#  * shortfin
+# get pinned to the corresponding nighly version. For stable releases,
+# * iree-base-compiler
+# * iree-base-runtime
+# * iree-turbine
+# * sharktank
+# * shortfin
+# get pinned to the corresponding `X.Y.*` version.
```

Comment:
Ooo this is cool. I hope it does what we want/expect though. Since a nightly/prerelease package has different pins than a stable release, we can't directly promote the nightly meta package to stable. We can design the release scripting to account for that, but any developers running scripts manually will have to watch for that.

Reply:
Fair point. But as it is a python only package without any real content I assume we can just build a new package?

Reply:
Yep. Easy enough to build locally for the desired configuration, but that does introduce a point where what we push to pypi is not what we run automated tests on. Hopefully that's fine for the metadata-only package.